pkg/controller/mutators/instances/mutator_controllers.go

```diff
@@ -34,7 +34,7 @@ func (a *Adder) Add(mgr manager.Manager) error {
 	// events is shared across all mutators that can affect the implied schema
 	// of kinds to be mutated, since these mutators can set each other into conflict
 	events := make(chan event.GenericEvent, eventQueueSize)
-	eventsSource := &source.Channel{Source: events}
+	eventsSource := &source.Channel{Source: events, DestBufferSize: 1024}
```

Comment:
why do we need to explicitly set this if default is already `1024`? from https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/source:
> 	// DestBufferSize is the specified buffer size of dest channels.
	// Default to 1024 if not specified.

Reply:
If we not set a value for this field, there will have a race condition when start multiple controllers:

https://github.com/open-policy-agent/gatekeeper/blob/19c9cef1cfc24a3f9b8b6fc3a593102788299796/vendor/sigs.k8s.io/controller-runtime/pkg/source/source.go#L251-L254

```
WARNING: DATA RACE
Read at 0x00c00004f118 by goroutine 226:
  sigs.k8s.io/controller-runtime/pkg/source.(*Channel).Start()
      /go/src/.../vendor/sigs.k8s.io/controller-runtime/pkg/source/source.go:252 +0xd3

Previous write at 0x00c00004f118 by goroutine 335:
  sigs.k8s.io/controller-runtime/pkg/source.(*Channel).Start()
      /go/src/.../vendor/sigs.k8s.io/controller-runtime/pkg/source/source.go:253 +0xec

```

Reply:
Basically, by setting a number, we are avoiding calling the setting of DestBufferSize on line 253

Reply:
That makes sense. So subtle!